### PR TITLE
[flutter_tools] Add more info to pub get failure event

### PR DESCRIPTION
**Pull request by zanderso**
## Description

Report whether an invocation of pub emits 'version solving failed' to analytics. Rework the pub events to capture the result of more invocations along with the context.

## Tests

I added the following tests:

New tests in pub_get_test.dart

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
lgtm

File: packages/flutter_tools/test/general.shard/dart/pub_get_test.dart [hunk 1]
```diff
@@ -144,14 +147,67 @@ void main() {
...
+
+  testUsingContext('analytics sent on success', () async {
+    MockDirectory.findCache = true;
+    await pubGet(context: PubContext.flutterTests, checkLastModified: false);
+    verify(flutterUsage.sendEvent('pub', 'flutter-tests', label: 'success')).called(1);
+  }, overrides: <Type, Generator>{
+    ProcessManager: () => MockProcessManager(0),
+    FileSystem: () => MockFileSystem(),
+    Platform: () => FakePlatform(
+      environment: <String, String>{'PUB_CACHE': 'custom/pub-cache/path'},
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/dart/pub_get_test.dart]: I would make this unmodifiable maps to reflect the Platform API, unless the FakePlatform already handles thiis

**Review by jonahwilliams — APPROVED** [review]
LGTM with nit

File: packages/flutter_tools/test/general.shard/dart/pub_get_test.dart (hunk 1 above)
Comment (zanderso) [reply packages/flutter_tools/test/general.shard/dart/pub_get_test.dart]: Done

**Pull request merged**